From 0cc3cc7185ff19f01014bcc26bb7c44b1c5b32f2 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Tue, 5 Mar 2019 19:23:06 -0500
Subject: [PATCH 6/6] aufs4: fix access_ok calls

Commit 96d4f267e [Remove 'type' argument from access_ok() function]
changed the access_ok calls to no longer need a type parameters. We
adjust the aufs calls to match.

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 fs/aufs/branch.c | 2 +-
 fs/aufs/mvdown.c | 2 +-
 fs/aufs/sysfs.c  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/aufs/branch.c b/fs/aufs/branch.c
index 598cf370a1dc..1f783609e634 100644
--- a/fs/aufs/branch.c
+++ b/fs/aufs/branch.c
@@ -1127,7 +1127,7 @@ static int au_ibusy(struct super_block *sb, struct aufs_ibusy __user *arg)
 
 	err = copy_from_user(&ibusy, arg, sizeof(ibusy));
 	if (!err)
-		err = !access_ok(VERIFY_WRITE, &arg->h_ino, sizeof(arg->h_ino));
+		err = !access_ok(&arg->h_ino, sizeof(arg->h_ino));
 	if (unlikely(err)) {
 		err = -EFAULT;
 		AuTraceErr(err);
diff --git a/fs/aufs/mvdown.c b/fs/aufs/mvdown.c
index 59ce2d700ec9..0b065e6fcfe9 100644
--- a/fs/aufs/mvdown.c
+++ b/fs/aufs/mvdown.c
@@ -643,7 +643,7 @@ int au_mvdown(struct dentry *dentry, struct aufs_mvdown __user *uarg)
 
 	err = copy_from_user(&args->mvdown, uarg, sizeof(args->mvdown));
 	if (!err)
-		err = !access_ok(VERIFY_WRITE, uarg, sizeof(*uarg));
+		err = !access_ok(uarg, sizeof(*uarg));
 	if (unlikely(err)) {
 		err = -EFAULT;
 		AuTraceErr(err);
diff --git a/fs/aufs/sysfs.c b/fs/aufs/sysfs.c
index 927f6b252c75..e939d0208a99 100644
--- a/fs/aufs/sysfs.c
+++ b/fs/aufs/sysfs.c
@@ -247,7 +247,7 @@ static int au_brinfo(struct super_block *sb, union aufs_brinfo __user *arg)
 
 	sz = sizeof(*arg) - offsetof(union aufs_brinfo, path);
 	for (bindex = 0; bindex <= bbot; bindex++, arg++) {
-		err = !access_ok(VERIFY_WRITE, arg, sizeof(*arg));
+		err = !access_ok(arg, sizeof(*arg));
 		if (unlikely(err))
 			break;
 
-- 
2.19.1

